src/lib/components/toast/useMutationsHandler.test.tsx

```diff
@@ -0,0 +1,131 @@
+import { act, renderHook } from '@testing-library/react-hooks';
+import { MutationConfig, useMutationsHandler } from './useMutationsHandler';
+import { useToast } from './ToastProvider';
+
+jest.mock('./ToastProvider', () => ({
+  useToast: jest.fn(),
+}));
+
+const mockUseToast = useToast as jest.MockedFunction<typeof useToast>;
+
+describe('useMutationsHandler', () => {
+  const mainMutation = {
+    mutation: {
+      isLoading: false,
+      isSuccess: true,
+      isError: false,
+      isIdle: false,
+    },
+    name: 'mutation1',
+  } as MutationConfig<unknown, unknown>;
+  const dependantMutations = [
+    {
+      mutation: {
+        isLoading: false,
+        isSuccess: true,
+        isError: false,
+        isIdle: false,
+      },
+      name: 'mutation2',
+    },
+    {
+      mutation: {
+        isLoading: false,
+        isSuccess: false,
+        isError: false,
+        isIdle: true,
+      },
+      name: 'mutation3',
+      isPrimary: false,
+    },
+  ] as MutationConfig<unknown, unknown>[];
```

Comment:
I think the generics should remain on the mutation level, we can have several mutation in this array and each of them with their own variable and return type